File: client/src/test/java/com/jraska/github/client/push/RemoteMessageToActionConverterTest.java
package com.jraska.github.client.push;

import com.jraska.github.client.common.Maps;

import org.junit.Test;

import java.util.Collections;
import java.util.HashMap;
import java.util.Map;

import static org.assertj.core.api.Assertions.assertThat;

public class RemoteMessageToActionConverterTest {
  @Test
  public void whenRemoteMessageWithoutAction_thenDefaultReturned() {
    PushAction action = RemoteMessageToActionConverter.convert(Collections.emptyMap());

    assertThat(action.name).isEqualTo(PushAction.DEFAULT.name);
  }

  @Test
  public void whenRemoteMessageWithAction_thenActionReturned() {
    PushAction action = RemoteMessageToActionConverter.convert(Maps.newHashMap("action", "heeeya"));

    assertThat(action.name).isEqualTo("heeeya");
    assertThat(action.parameters).isEmpty();
  }

  @Test
  public void whenRemoteMessageWithParameters_thenActionReturned() {
    Map<String, String> map = new HashMap<>();
    map.put("action", "heeeya");
    map.put("parameter", "true");

    PushAction action = RemoteMessageToActionConverter.convert(map);

    assertThat(action.name).isEqualTo("heeeya");
    assertThat(action.parameters).hasSize(1);
    assertThat(action.parameters.get("parameter")).isEqualTo("true");
  }
}

File: client/src/test/java/com/jraska/github/client/push/RemoteMessageToActionConverterTest.kt
package com.jraska.github.client.push

import org.assertj.core.api.Assertions.assertThat
import org.junit.Test
import java.util.Collections

class RemoteMessageToActionConverterTest {
  @Test
  fun whenRemoteMessageWithoutAction_thenDefaultReturned() {
    val action = RemoteMessageToActionConverter.convert(Collections.emptyMap<String, String>())

    assertThat(action.name).isEqualTo(PushAction.DEFAULT.name)
  }

  @Test
  fun whenRemoteMessageWithAction_thenActionReturned() {
    val action = RemoteMessageToActionConverter.convert(mapOf("action" to "heeeya"))

    assertThat(action.name).isEqualTo("heeeya")
    assertThat(action.parameters).isEmpty()
  }

  @Test
  fun whenRemoteMessageWithParameters_thenActionReturned() {
    val map = mapOf("action" to "heeeya", "parameter" to "true")

    val action = RemoteMessageToActionConverter.convert(map)

    assertThat(action.name).isEqualTo("heeeya")
    assertThat(action.parameters).hasSize(1)
    assertThat(action.parameters["parameter"]).isEqualTo("true")
  }
}
